File: Data Structures & Algorithms/path-sum/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        
        def validSum(node, currSum):
            if not node:
                return False

            pathSum = currSum + node.val

            if (pathSum == targetSum) and (not node.left) and (not node.right):
                return True
            
            if validSum(node.left, pathSum):
                return True
            if validSum(node.right,pathSum):
                return True

            return False

        return validSum(root, 0)

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        traverse_helper(root);
        return postOrder;
    }

    void traverse_helper(TreeNode* root){
        if(root == nullptr)
            return;

        traverse_helper(root->left);
        traverse_helper(root->right);
        postOrder.push_back(root->val);

    }

    vector<int> postOrder;
};